handbook/repository/aws-secrets.md

```diff
@@ -164,6 +172,23 @@ Are you sure [Y/n]? # [enter] to confirm
 It's recommended to use `SecureString` in most cases. However, if you need to add an email address, or an email sender's name to the secrets, you can just use a `String`.
 {% endhint %}
 
+{% hint style="info" %}
+Note that this command only creates the secret in one AWS account at a time (eg `island-is-development01`). To create a secret in all environments, you need to run it with each corresponding AWS account configured, by going through the steps in [Using AWS session](#using-aws-session) multiple times.
+
+To make this easier we recommend configuring SSO for each AWS account using a different AWS profile (dev, staging, prod). Then you can create a secret in all environments like this:
+
+```bash
+aws sso login --profile dev
+aws sso login --profile staging
+aws sso login --profile prod
```

Comment:
You can skip this step and improve it by configuring the config with:
```
# ~/.aws/config

[profile islandis-dev-sso]
sso_start_url = https://island-is.awsapps.com/start
sso_account_id = 013313053092
sso_role_name = AWSPowerUserAccess
sso_region = eu-west-1

[profile islandis-staging-sso]
sso_start_url = https://island-is.awsapps.com/start
sso_account_id = 261174024191
sso_role_name = Secret_Service
sso_region = eu-west-1

[profile islandis-prod-sso]
sso_start_url = https://island-is.awsapps.com/start
sso_account_id = 251502586493
sso_role_name = Secret_Service
sso_region = eu-west-1
```

and then in credentials, it'll automatically load them using aws-vault:
```
# ~/.aws/credentials

[islandis-dev]
credential_process = aws-vault exec islandis-dev-sso --json

# ... do the same thing for staging and prod
```


Now, if you want to run any of the scripts related to AWS, you only have to do:
```
AWS_PROFILE=islandis-dev yarn create-secret
```

No need to manually login (if your session has expired, you'll be navigated to the sso page automatically)

Reply:
Hm. Interesting. Do you know if the JS aws-sdk v3 supports credential process in the default credential resolution?

I'm not sure if I want to require one more dependency (aws-vault) just for this. I actually think it might be enough to run `aws sso login --profile islandis-dev-sso` for all profiles to work, assuming they share an SSO access token.

Looks like AWS Vault is ideal to improve security for long lasting credential, by storing them in the keychain rather than in files. I don't have experience with it so I think it's outside the scope of this PR.

Though I think it could be a useful `{% hint info %}` if you feel like adding it.

Reply:
It is kinda nice to get that `~/.aws/config` you posted which can just be copy pasted to skip a bunch of manual steps and "pointless decisions".

Should we just post it in the docs? Can't imagine it's too sensitive. Or if it will change more frequently than we would update these docs anyways.

Interested in @pshomov's opinion on all of this.

Reply:
Yeah agree ... the account_id is not sensitive, nor the name of the sso role

Reply:
Sorry, I have not used aws-vault but have this in my .zshrc:
```
export AWS_DEFAULT_REGION=eu-west-1
export AWS_PAGER=
function sso (){
  export AWS_PROFILE=$1
  aws-sso-util login --profile $AWS_PROFILE
  $(aws-export-credentials --profile $AWS_PROFILE --credentials-file-profile $AWS_PROFILE)
}
```

And when the credentials expire, I just run it again.